ion-schema/src/isl/ranges.rs

```diff
@@ -379,50 +337,22 @@ mod base {
         }
     }
 
-    impl<T: Into<Element> + Clone + PartialEq> WriteToIsl for &Range<T> {
-        fn write_to<W: IonWriter>(&self, writer: &mut W) -> IonSchemaResult<()> {
+    impl<T: PartialEq + Display> Display for Range<T> {
+        fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
             match &self.lower {
-                Limit::Inclusive(value) if self.lower == self.upper => {
-                    writer.write_element(&value.clone().into())?;
-                    Ok(())
-                }
-                _ => {
-                    writer.set_annotations(["range"]);
-                    writer.step_in(IonType::List)?;
-                    match &self.lower {
-                        Limit::Unbounded => writer.write_symbol("min")?,
-                        Limit::Inclusive(value) => writer.write_element(&value.clone().into())?,
-                        Limit::Exclusive(value) => {
-                            writer.set_annotations(["exclusive"]);
-                            writer.write_element(&value.clone().into())?;
-                        }
-                    }
-                    match &self.upper {
-                        Limit::Unbounded => writer.write_symbol("max")?,
-                        Limit::Inclusive(value) => writer.write_element(&value.clone().into())?,
-                        Limit::Exclusive(value) => {
-                            writer.set_annotations(["exclusive"]);
-                            writer.write_element(&value.clone().into())?;
-                        }
-                    }
-                    writer.step_out()?;
-                    Ok(())
-                }
+                Limit::Inclusive(value) if self.lower == self.upper => value.fmt(f),
+                _ => f.write_fmt(format_args!("range::[{},{}]", self.lower, self.upper)),
```

Comment:
This is my first time seeing a manual invocation of `format_args!` in the wild. 🤔